### DISCUSSION TITLE
[ 已采纳 ] [ 发布 - 包文件 ] 能不能生成asar文件

### DISCUSSION THREAD

**Discussion by xmengnet** [GeekLee2012/Less-Player-Desktop#8] [General]
能不能生成单独的asar文件，这样就可以使用系统的electron了，不需要每次都那么大一个包了

**Comment by GeekLee2012** [comment]
这个建议看着应该可行，但我还没有测试过，具体操作也不确定。
一般来说，单独的asar文件，对普通用户来说，可能并不是很友好。
而为开发者提供的话，也并不是很必要。

这种方式，我感觉更适合做成版本自动升级更新（包括增量更新），
而不是提供asar文件让用户下载。

**Comment by xmengnet** [comment]
有包管理器的会方便很多，在linux和mac都很方便，我有在用有一个直接生成asar的[网易云音乐客户端](https://github.com/Rocket1184/electron-netease-cloud-music) https://github.com/Rocket1184/electron-netease-cloud-music 挺好用的

**Comment by GeekLee2012** [comment]
asar包已在国内Gitee平台提供，
由于包大小有些大，50M+，后续还是暂时在国内平台提供。

仅讨论本项目，正常包大小，应该是现在的一半左右。
由于没有使用Webpack等相关构建工具，且部分依赖太多，
导致除去资源等文件，依赖自身也有37M左右......

暂时先这样吧，也不打算去提供包管理器方式。
